python/lsst/afw/math/_function.cc

```diff
@@ -0,0 +1,152 @@
+                cls.def("toString", &Function2<ReturnT>::toString, "prefix"_a = "");
+                cls.def("getDFuncDParameters", &Function2<ReturnT>::getDFuncDParameters, "x"_a, "y"_a);
+            });
+}
+
+template <typename ReturnT>
+void declareBasePolynomialFunction2(lsst ::utils::python::WrapperCollection &wrappers, const std::string &suffix) {
+    auto const name = "BasePolynomialFunction2" + suffix;
+    using PyClass=py::class_<BasePolynomialFunction2<ReturnT>, std::shared_ptr<BasePolynomialFunction2<ReturnT>>,Function2<ReturnT>>;
+    wrappers.wrapType(PyClass(wrappers.module,name.c_str()),
+            [](auto &mod,auto &cls) {
+                cls.def("getOrder", &BasePolynomialFunction2<ReturnT>::getOrder);
+                cls.def("isLinearCombination", &BasePolynomialFunction2<ReturnT>::isLinearCombination);
+                cls.def_static("nParametersFromOrder", &BasePolynomialFunction2<ReturnT>::nParametersFromOrder,
+                               "order"_a);
+                cls.def_static("orderFromNParameters", &BasePolynomialFunction2<ReturnT>::orderFromNParameters,
+                               "nParameters"_a);
+                cls.def("getDFuncDParameters", &BasePolynomialFunction2<ReturnT>::getDFuncDParameters, "x"_a, "y"_a);
+            });
+}
+
+template <typename ReturnT>
+void declareNullFunction1( lsst::utils::python::WrapperCollection &wrappers, const std::string &suffix) {
+    auto const name = "NullFunction1" + suffix;
+    using PyClass=py::class_<NullFunction1<ReturnT>, std::shared_ptr<NullFunction1<ReturnT>>, Function1<ReturnT>>;
+    wrappers.wrapType(PyClass(wrappers.module,name.c_str()),
+            [](auto &mod,auto &cls) {
+                cls.def(py::init<>());
+
+                cls.def("clone", &NullFunction1<ReturnT>::clone);
+            });
+}
+
+template <typename ReturnT>
+void declareNullFunction2( lsst::utils::python::WrapperCollection &wrappers, const std::string &suffix) {
+    auto const name = "NullFunction2" + suffix;
+    using PyClass=py::class_<NullFunction2<ReturnT>, std::shared_ptr<NullFunction2<ReturnT>>, Function2<ReturnT>>;
+    wrappers.wrapType(PyClass(wrappers.module,name.c_str()),
+            [](auto &mod,auto &cls) {
+                cls.def(py::init<>());
+                cls.def("clone", &NullFunction2<ReturnT>::clone);
+            });
+}
+
+template <typename ReturnT>
+void declareAllFunctions(lsst::utils::python::WrapperCollection &wrappers, const std::string &suffix) {
+    declareFunction<ReturnT>(wrappers, suffix);
+    declareFunction1<ReturnT>(wrappers, suffix);
+    declareFunction2<ReturnT>(wrappers, suffix);
+    declareBasePolynomialFunction2<ReturnT>(wrappers, suffix);
+    declareNullFunction1<ReturnT>(wrappers, suffix);
+    declareNullFunction2<ReturnT>(wrappers, suffix);
+};
+
+}  // namespace <anonymous>
+
+void wrapFunction(lsst::utils::python::WrapperCollection &wrappers) {
+    declareAllFunctions<float>(wrappers, "F");
```

Comment:
Needs `addSignatureDependency("lsst.afw.table.io")` to support `addPersistableMethods`.